### ISSUE TITLE
iOS 14 Toggle

### ISSUE THREAD

**Issue by martinrz2013** [adjust/unity_sdk#190]
Hi,
We are integrating the latest version 4.23.0 with better support for iOS 14.
There is a toggle to enable it, but it only sets the flag in EditorPrefs.
We need a persistent solution that can be used across multiple machines and Unity instances.
A better approach would be to save it in a settings file that can be pushed to the repository and read by the post processor.

As a workaround we are enabling it always in the AdjustEditor file, but it would be great to have this already handled by the SDK.

Thank you!

**Comment by shinji19** [comment]
+1

**Comment by wesclarkemag** [comment]
Hello. We've just integrated 4.23.0 into our solution as well and were surprised to find this setting stored locally: this really needs to be stored at the project level and shared across source control. Many Unity projects are team based affairs these days with CI producing builds. @martinrz2013 suggestion sounds like a good approach, and you could probably move the rest of Adjusts settings there too, and not have them stored in a prefab (bad for merging unless you break the association). Thanks.

**Comment by fliperamma** [comment]
Hello team. Any update on this? 
I like the idea @martinrz2013 suggested, isolating it from the library. Perhaps you can follow a similar implementation as Facebook did for their `FacebookSettings` asset:
[FacebookSettings.cs](https://github.com/facebook/facebook-sdk-for-unity/blob/master/Facebook.Unity.Settings/FacebookSettings.cs)
[FacebookSettingsEditor.cs](https://github.com/facebook/facebook-sdk-for-unity/blob/master/Facebook.Unity.Editor/FacebookSettingsEditor.cs)

**Comment by ugi** [comment]
Hey. Thanks for pointing this out, pretty useful. And yeah, we piled up few things we wanna ship with next Unity SDK patch update and we're probably gonna start working on this during this week. Patch update to be expected either end of this week or (more realistically) during next week. Will keep you guys posted.

**Comment by fliperamma** [comment]
That will be very nice, we have a few releases this week, so the sooner you guys get this out the better for us (testing etc) ofc.

Another question please @ugi: If we want to release with **Xcode 11.6**, we'd need to disable the iOS 14 support, would this be the same as using `[4.22.1](https://github.com/adjust/unity_sdk/releases/tag/v4.22.1)`?

**Comment by ugi** [comment]
Hey @fliperamma 

Sorry for a bit of delay. Update will probably happen early next week.

And as far as v4.22.1 is concerned, that version shouldn't even contain the toggle for iOS 14 support since we added that one in v4.23.0.

**Comment by fliperamma** [comment]
While we don't have the actual solution in place for this toggle issue, another possible workaround could be to do something like this in your Build Script / Pre Process. If building via CI, you could also create an environment variable and read it in your Build Script. 

```c#
EditorPrefs.SetBool("adjustiOS14Support", someEnvironmentVariable);
```

**Comment by ugi** [comment]
@martinrz2013 and rest of interested people,

I made this [as suggested](https://github.com/adjust/unity_sdk/issues/190#issuecomment-678173839) and pretty much how Facebook did it with their **FacebookSettings.cs**. Before making the release, one question though (trying to understand best practice): Would you really expect **AdjustSettings.asset** to be created in **Assets/Adjust/Editor** folder or is there maybe some more meaningful place for it? Facebook is putting it into [Resources folder](https://github.com/facebook/facebook-sdk-for-unity/blob/master/Facebook.Unity.Settings/FacebookSettings.cs#L35), so just double checking if suggested location is fine to go with or maybe you would like to see this asset elsewhere.

**Comment by fliperamma** [comment]
Hey, @ugi thanks for getting the next release ready.

Unity doesn't recommend using the Resources folder as it is common for devs to abuse it, which might cause issues as described [here](https://docs.unity3d.com/Manual/BestPracticeUnderstandingPerformanceInUnity6.html). This being said, having one single asset from Adjust there should be completely fine.

Others might chip in, but I personally do like the way Facebook solved it as the Settings can be accessed _directly_, both at Runtime and in the Editor, which I believe provides more flexibility to them and to us (e.g. The [`FB.cs`](https://github.com/facebook/facebook-sdk-for-unity/blob/master/Facebook.Unity/FB.cs#L187) accesses it at Runtime, we can also access it on an Editor BuildScript/Pre Process to, for example, change which App Id we'll use depending on which environment we want to build).

I'm not sure how the changes are looking like but, as you probably know, right now people using CI to build are probably making changes via the **Adjust prefab** in their Build Scripts (e.g. set the `Adjust.environment`, `Adjust.logLevel` etc per environment being built) and now those will need to point to the **AdjustSettings asset** instead (if I understood it correctly). No big deal I think, but just something to be mindful and perhaps highlight in the Changelog as it is a breaking change for those use cases.

**Comment by ugi** [comment]
Hi @fliperamma 

And thanks for the comments. Actually, we were not planning to touch anything prefab related and move it to **AdjustSettings.asset**. We only intended to move two boolean variables we have in **AdjustEditor.cs** - the one indicating if post processing scripts should take place or not and the one indicating if app should be built with iOS 14 support or not.

Maybe for sake of discussion (we're putting things to public repo pretty much once we have finalised the release thus, files in here), here's how I intended **AdjustEditor.cs** and **AdjustSettings.cs** to look like. I was not planning to add **AdjustSettings.asset** with some default values to the **unitypackage** since this is something variable and I am not sure if SDK updates with having always default values in there will conflict with your custom settings each time you update the SDK (I assume they will). **AdjustSettings.cs** is in charge to create asset in **Assets/Adjust/Editor** folder in case asset doesn't exist and I guess it should be okay from that moment on.

Lemme know if code, at least conceptually makes sense to you. You might as well test it by simply trying out these two files if you're interested. Any feedback is more than welcomed.

[AdjustEditorFolderContent.zip](https://github.com/adjust/unity_sdk/files/5282094/AdjustEditorFolderContent.zip)

GitHub doesn't allow upload of **.cs** files, so **.zip**.

**Comment by ugi** [comment]
Hey guys,

We just released [v4.23.1](https://github.com/adjust/unity_sdk/releases/tag/v4.23.1) with suggested changes. Feel free to give it a shot and reopen/comment in case you encounter any issues or unexpected behaviour.

Thank you one more time for the idea and advices.

Cheers!
